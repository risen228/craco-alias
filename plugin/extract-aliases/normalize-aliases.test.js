/* eslint-disable sonarjs/no-duplicate-string */
const path = require('path')

const normalizeAliases = require('./normalize-aliases')

describe('normalize-aliases', () => {
  const appPath = path.resolve(__dirname, '../..')

  test('should correctly normalize aliases', () => {
    expect(
      normalizeAliases({
        absoluteBaseUrl: path.resolve(appPath, '.'),
        aliases: {
          '@file': './src/file.js',
          '@file2': 'src/file2.js',
          '@dir': './src/dir',
          '@dir2': '././src/dir2/',
          '$dir3': 'src/dir3',
          'my-package': './node_modules/some-package',
          'external-package': '/absolute_path/external-package',
          '@material-ui': 'node_modules/@material-ui/ie-10/ie-10.js',
        },
      })
    ).toMatchSnapshot()

    expect(
      normalizeAliases({
        absoluteBaseUrl: path.resolve(appPath, './src'),
        aliases: {
          '@file': './file.js',
          '@file2': 'file2.js',
          '@dir': './dir',
          '@dir2': '././dir2/',
          '$dir3': 'dir3',
        },
      })
    ).toMatchSnapshot()

    expect(
      normalizeAliases({
        absoluteBaseUrl: path.resolve(appPath, './fallback_test'),
        aliases: {
          '$dir3': ['dir3', 'fallback/dir3'],
          '@file2': ['file2.js'],
        },
      })
    ).toMatchSnapshot()
  })
})
