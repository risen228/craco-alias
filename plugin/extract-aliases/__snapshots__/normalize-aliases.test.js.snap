// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`normalize-aliases should correctly normalize aliases 1`] = `
Object {
  "$dir3": "<PROJECT_ROOT>/src/dir3",
  "@dir": "<PROJECT_ROOT>/src/dir",
  "@dir2": "<PROJECT_ROOT>/src/dir2",
  "@file": "<PROJECT_ROOT>/src/file.js",
  "@file2": "<PROJECT_ROOT>/src/file2.js",
  "@material-ui": "<PROJECT_ROOT>/node_modules/@material-ui/ie-10/ie-10.js",
  "external-package": "/absolute_path/external-package",
  "my-package": "<PROJECT_ROOT>/node_modules/some-package",
}
`;

exports[`normalize-aliases should correctly normalize aliases 2`] = `
Object {
  "$dir3": "<PROJECT_ROOT>/src/dir3",
  "@dir": "<PROJECT_ROOT>/src/dir",
  "@dir2": "<PROJECT_ROOT>/src/dir2",
  "@file": "<PROJECT_ROOT>/src/file.js",
  "@file2": "<PROJECT_ROOT>/src/file2.js",
}
`;

exports[`normalize-aliases should correctly normalize aliases 3`] = `
Object {
  "$dir3": Array [
    "<PROJECT_ROOT>/fallback_test/dir3",
    "<PROJECT_ROOT>/fallback_test/fallback/dir3",
  ],
  "@file2": Array [
    "<PROJECT_ROOT>/fallback_test/file2.js",
  ],
}
`;
